### PR TITLE
Add a web debugger

### PR DESCRIPTION
**Pull request by remicalixte**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import "flag"
+
+var (
+	debugServer bool
+	debugPort   int
+)
+
+func init() {
+	flag.BoolVar(&debugServer, "debug-server", false, "enable the debug server")
```
Comment (sfluor) [inline config/config.go]: should we only use `debug-port` and only enable the debug server if the port is provided ?

Comment (remicalixte) [reply config/config.go]: We wouldn't be able to use a default value for the port then

File: debugger/debugger.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+//Start the debugger and open the web page
+func (db *Debugger) Start() {
+	go func() {
+		err := db.s.ListenAndServe()
+		fmt.Println(err)
+	}()
+
+	fmt.Printf("open web browser at %s\n", db.addr)
+	cmd := exec.Command("xdg-open", fmt.Sprintf("http://%s", db.addr))
```
Comment (sfluor) [inline debugger/debugger.go]: I think we should first check if `xdg-open` is available (if not we can just log the URL that should be opened or try to fallback on `open`).

File: main.go [hunk 3]
```diff
@@ -4,22 +4,40 @@ import (
...
 	emu := core.New()
 	emu.ReadROM(flag.Arg(0))
-	emu.CPU.Start()
+	emu.Start()
+
+	if config.DebugServer() {
+		fmt.Println("start debugger")
+		db := debugger.New(emu, fmt.Sprintf("localhost:%d", config.DebugPort()))
+		db.Start()
+
```
Comment (sfluor) [inline main.go]: nit: white line

**Pull request merged**